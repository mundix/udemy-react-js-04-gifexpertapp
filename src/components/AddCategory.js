import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    // Necesito saber lo que l apersona escribe
    const [inputValue, setInputValue] = useState();

    // Cada ves que esa caja de trexto cambie, se ejecuta esta funcion de flecha
    const handleInputChange = (e) => {
        const value = e.target.value;
        setInputValue(value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.length > 2) {
            setCategories(cats => [...cats, inputValue]);
            setInputValue('');
        }
        // console.log('Submit Hecho');
    }

    //Si el form es el unico elemento que se devuelve ,no neceisto el <> los fragments </>
    return (
        <form onSubmit={handleSubmit}>
            <h1>{inputValue}</h1>
            <input 
                type="text" 
                value = {inputValue}
                // onChange={handleInputChange}
                onChange={ (e) => {handleInputChange(e);}}
                />   
        </form>        
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
  };