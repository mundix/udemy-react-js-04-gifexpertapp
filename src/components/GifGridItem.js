import React from 'react';

// export const GifGridItem = ( props) => {
export const GifGridItem = ( {id, title, url}) => {


    return (
        <div className='card'>
            {/* {img.title} */}
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
