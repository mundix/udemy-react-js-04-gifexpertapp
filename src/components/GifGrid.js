import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    // Necesito recibir la categoria y llamar la peticion http 
    const [images, setImages] = useState([]); //inicializamos images como array vacio

    useEffect(() => {
        getGifs();
    }, []); //cuando se pone el [] , ecita que se ejecute mas de una vez 

    const getGifs = async () => {
        const url = `https://api.giphy.com/v1/gifs/search?api_key=Nr4bfrgRK7obHM3O68YwvQ2iUmuxi8Yj&q=Samurai+X&limit=10`;
        const resp = await fetch(url);
        // const data = await resp.json();
        const { data } = await resp.json();
        // resp.ok ?

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                // url: img.images.downsized_medium.url
                url: img.images?.downsized_medium.url //? es por si no llega 
            };
        });
        console.log(gifs);
        setImages(gifs);
    }

    // getGifs();

    return (
        <>
            <h3>{category}</h3>
            <div className='card-grid'>
                {/* <ol> */}

                {/* {images.map( images => {
                    return <li key={images.id}>{images.title}</li>;
                })} */}
                {/* Return implicit */}
                {/* {images.map( img => (
                    <li key={img.id}>{img.title}</li>
                ))} */}
                {/* Usando la desestructuracion */}
                {/* {images.map( ({id, title}) => ( */}
                {images.map(img => (
                    <GifGridItem
                        key={img.id}
                        // img={img}
                        // Mando cada una de las propiedades en ves de la img con sus propiedeaes
                        {...img}
                    />
                ))}

                {/* </ol> */}
            </div>
        </>

    )
}
