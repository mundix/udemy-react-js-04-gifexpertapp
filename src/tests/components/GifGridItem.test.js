import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas en archivo GifGridItem', () => {

   const title = 'Un titulo';
   const url = 'https://localhost/algo.js';
    
   test('Comprobar GifGridItem correctamente', () => {
       
        const wrapper = shallow(<GifGridItem title={title} url={url}/>);
        expect(wrapper).toMatchSnapshot();
   })
    
});