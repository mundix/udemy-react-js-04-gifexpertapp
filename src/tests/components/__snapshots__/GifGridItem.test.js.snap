// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en archivo GifGridItem Comprobar GifGridItem correctamente 1`] = `
<div
  className="card animate__animated  animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.js"
  />
  <p>
    Un titulo
  </p>
</div>
`;
